### PR TITLE
updating troubleshooting and upgrade changelog

### LINKED ISSUE #8090: Update Loki configuration doc `querier.query_timeout` -> `limits_config.query_timeout`
**Describe the bug**
Since (version to be confirmed) https://grafana.com/docs/loki/latest/configuration/ is no longer correct.
To increase `query_timeout` from the 1m default, the option has to be added to `limits_config`.
The current documentation still references the `querier` config block.

**To Reproduce**
1. [Configuration migration PR](https://github.com/grafana/loki/pull/7471/files) indicates the new configuration property to use.

**Expected behavior**
Docs need to update references to `querier``query_timeout` to `limits_config``query_timeout`.

**Environment:**
 - All

**Screenshots, Promtail config, or terminal output**
see https://grafana.com/docs/loki/latest/configuration/#querier

### PR DESCRIPTION
**Pull request by yashsoni369**
**What this PR does / why we need it**:
Updating docs which mentions `querier.query_timeout` to `limits_request.query_timeout`

**Which issue(s) this PR fixes**:
Fixes #8090 

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: docs/sources/upgrading/_index.md [hunk 1]
```diff
@@ -103,11 +103,11 @@ These statistics are also displayed when using `--stats` with LogCLI.
 The new `push` mode to [Loki canary](https://grafana.com/docs/loki/latest/operations/loki-canary/) can push logs that are generated by a Loki canary directly to a given Loki URL. Previously, it only wrote to a local file and you needed some agent, such as promtail, to scrape and push it to Loki.
 So if you run Loki behind some proxy with different authorization policies to read and write to Loki, then auth credentials we pass to Loki canary now needs to have both `READ` and `WRITE` permissions.
 
-### Engine query timeout is deprecated
+### Engine query timeout and Querier `query_timeout` is deprecated
```
Comment (chaudum) [inline docs/sources/upgrading/_index.md:106]: I think it makes sense to name the actual settings:

```suggestion
### `engine.timeout` and `querier.query_timeout` are deprecated
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 #### Loki
 
 ##### Enhancements
+* [8206](https://github.com/grafana/loki/pull/8206) **yashsoni369**: Updating doc removing `querier.query_timeout` to `limits_requests.query_timeout`.
```
Comment (chaudum) [inline CHANGELOG.md:8]: nit: Docs changes do not require a changelog entry.

**Review by chaudum — COMMENTED** [review]
Thanks for taking the time and effort to improve/fix the upgrade documentation. I left some notes.

File: CHANGELOG.md (hunk 2 above)
Comment (yashsoni369) [reply CHANGELOG.md]: Noted

Comment (yashsoni369) [reply CHANGELOG.md]: Updated @chaudum

**Review by owen-d — APPROVED** [review]
Thanks!

**Pull request merged**